doc/developer/design/20230705_v2_txn_management.md

```diff
@@ -0,0 +1,264 @@
+# Platform V2: Txn Management
+
+## Context
+
+Our SQL Table Transactions Management still shows some of its lineage from the
+single-binary. At times, this conflicts with the "Platform V2" goals of
+_use-case isolation_ and _zero-downtime upgrades (0dt)_.
+
+## Goals
+
+Use-case isolation. Make it possible to prevent pgwire sessions from interfering
+with each other through:
+
+- Large reads
+- Large writes
+- Frequent reads
+- Frequent writes
+- Long-running transactions
+- WIP anything else?
+
+Zero-downtime upgrades. Nothing in particular motivating here. Don't introduce
+anything that interferes with 0dt.
+
+Don't break things that work today:
+- Explicit SQL transactions that are entirely reads or entirely writes (plus
+  "constant" reads)
+- Multi-table read transactions
+- Single-table write transactions
+- Reads include from `MATERIALIZED VIEW`s
+- Transactions that are implicitly reads followed by writes:
+  - `INSERT INTO ... SELECT FROM ...`
+  - `UPDATE`
+  - `DELETE`
+- Don't regress performance "too much". Moving to a distributed system is
+  necessary to provide isolation and distributed transactions are necessarily
+  less performant. However, WIP figure out some precise framing of what's
+  acceptable vs not.
+
+Nice to haves, while we're revisiting SQL tables and transactions:
+- A path to multi-table write transactions. It need not end at the global
+  optimum, just be possible.
+- Bounded memory usage
+- Highly optimized `COPY INTO`/`COPY FROM`
```

Comment:
This should maybe get replaced with "highly optimized webhooks", or at the least it should be added at a higher priority than tables.